client/src/sagas/publish.js

```diff
@@ -5,44 +5,77 @@ import { updateError, updatePublishStatus, clearFile } from '../actions/publish'
 import { selectPublishState } from '../selectors/publish';
 import { selectChannelState } from '../selectors/channel';
 import { selectSiteState } from '../selectors/site';
+import { selectShowState, selectAsset } from '../selectors/show';
 import { validateChannelSelection, validateNoPublishErrors } from '../utils/validate';
 import { createPublishMetadata, createPublishFormData, createThumbnailUrl } from '../utils/publish';
 import { makePublishRequestChannel } from '../channels/publish';
 
 function * publishFile (action) {
   const { history } = action.data;
-  const { publishInChannel, selectedChannel, file, claim, metadata, thumbnailChannel, thumbnailChannelId, thumbnail, error: publishToolErrors } = yield select(selectPublishState);
+  const publishState = yield select(selectPublishState);
+  const { publishInChannel, selectedChannel, file, claim, metadata, thumbnailChannel, thumbnailChannelId, thumbnail, isUpdate, error: publishToolErrors } = publishState;
   const { loggedInChannel } = yield select(selectChannelState);
   const { host } = yield select(selectSiteState);
+
+  let show, asset;
+  if (isUpdate) {
+    show = yield select(selectShowState);
+    asset = selectAsset(show);
+  }
   // validate the channel selection
   try {
     validateChannelSelection(publishInChannel, selectedChannel, loggedInChannel);
   } catch (error) {
     return yield put(updateError('channel', error.message));
-  };
+  }
   // validate publish parameters
   try {
     validateNoPublishErrors(publishToolErrors);
   } catch (error) {
     return console.log('publish error:', error.message);
   }
-  // create metadata
-  let publishMetadata = createPublishMetadata(claim, file, metadata, publishInChannel, selectedChannel);
-  if (thumbnail) {
-    // add thumbnail to publish metadata
-    publishMetadata['thumbnail'] = createThumbnailUrl(thumbnailChannel, thumbnailChannelId,  claim, host);
+
+  let publishMetadata, publishFormData, publishChannel;
+  if (isUpdate) {
+    publishMetadata = createPublishMetadata(asset.name, {type: asset.claimData.contentType}, metadata, publishInChannel, selectedChannel);
+    publishMetadata['channelName'] = asset.claimData.channelName;
+    if (thumbnail) {
+      // add thumbnail to publish metadata
+      publishMetadata['thumbnail'] = createThumbnailUrl(thumbnailChannel, thumbnailChannelId, claim, host);
+    }
+    // create form data for main publish
+    publishFormData = createPublishFormData(file, thumbnail, publishMetadata);
+    // make the publish request
+    publishChannel = yield call(makePublishRequestChannel, publishFormData, true);
```

Comment:
Can move this out as `publishChannel = yield call(makePublishRequestChannel, publishFormData, isUpdate);`